types/index.d.ts

```diff
@@ -0,0 +1,509 @@
+  off(event?: 'afterRender', callback?: OktaSignIn.EventCallback): void;
+
+  show(): void;
+  hide(): void;
+  remove(): void;
+
+  showSignInToGetTokens(options: OktaSignIn.ShowSignInToGetTokensOptions): Promise<Tokens>;
+  showSignInAndRedirect(options: OktaSignIn.ShowSignInAndRedirectOptions): Promise<void>;
+  renderEl(
+    options: OktaSignIn.RenderElOptions,
+    success?: (res: OktaSignIn.RenderResult) => void,
+    error?: (err: OktaSignIn.RenderError) => void
+  ): Promise<OktaSignIn.RenderResult>;
+}
+
+declare namespace OktaSignIn {
+  // Config
+  interface WidgetConfig {
+    // Basic config options
+    baseUrl: string;
+    logo?: string;
+    logoText?: string;
+    helpSupportNumber?: string;
+    brandName?: string;
+    // Username and password
+    username?: string;
+    transformUsername?: (username: string, operation: UserOperation) => string;
+    processCreds?: (creds: Creds, callback?: SimpleCallback) => void;
+    // Language and text
+    language?: LanguageCode | LanguageCallback;
+    i18n?: {
+      [languageCode in LanguageCode]?: {
+        [propertyKey: string]: string;
+      };
+    };
+    assets?: {
+      baseUrl?: string;
+      rewrite?: (assetPath: string) => string;
+    };
+    // Colors
+    colors?: {
+      [colorKey in ColorKey]?: string;
+    };
+    // Links
+    helpLinks?: {
+      help?: string;
+      forgotPassword?: string;
+      factorPage?: Link;
+      unlock?: string;
+      custom?: Array<Link>;
+    };
+    signOutLink?: string;
+    // Buttons
+    customButtons?: Array<CustomButton>;
+    // Registration
+    registration?: Registration.Callbacks;
+    policyId?: string;
+    // Feature flags
+    features?: {
+      [featureKey in Feature]?: boolean;
+    };
+    // OIDC
+    clientId?: string;
+    redirectUri?: string;
+    authParams?: AuthParams;
+    authClient?: OktaAuth;
+    oAuthTimeout?: number;
+    // IdP
+    idps?: Array<SocialIdp | CustomIdp>;
+    idpDisplay?: IdpDisplay;
+    idpDiscovery?: {
+      requestContext?: string;
+    };
+    // Smart Card IdP (X509)
+    piv?: Piv;
+    // Bootstrapping
+    recoveryToken?: string;
+    stateToken?: string;
+    relayState?: string;
+    // Callbacks
+    globalSuccessFn?: (res: RenderResult) => void;
+    globalErrorFn?: (res: RenderError) => void;
+    // IDX
+    apiVersion?: string;
+    // Consent
+    consent?: {
+      cancel?: SimpleCallback;
+    };
+  }
+
+  // Auth params
+  interface AuthParams extends OktaAuthOptions {
+    display?: Display;
+    responseType?: ResponseType | Array<ResponseType>;
+    responseMode?: ResponseMode;
+    scopes?: Array<Scope>;
+    state?: string;
+    nonce?: string;
+    authScheme?: string;
+  }
+  type Display =
+    'popup' |
+    'page';
+  type ResponseMode =
+    'okta_post_message' |
+    'fragment' |
+    'query' |
+    'form_post';
+  type ResponseType =
+    'code' |
+    'token' |
+    'id_token' |
+    'access_token';
```

Comment:
`access_token` doesn't seem to be listed in the API [reference](https://developer.okta.com/docs/reference/api/oidc/#request-parameters) is it intended for custom authorization servers?

Reply:
I've found `access_token` in [readme](https://github.com/okta/okta-signin-widget):
```
Valid response types are id_token, access_token, and code.
```

Reply:
Removed `access_token`. Looks like error in readme